### PR TITLE
[ENH] torch based MLP classifier

### PR DESCRIPTION
**Pull request by RecreationalMath**
<details>
<summary>Moved from draft to Review.</summary>
<strike>Draft PR. 
Opened to ask some specific questions.
The details below will be filled once ready for review. Test failures can be ignored till then.</strike>

</details>

#### Adds new MLP classifier in PyTorch as part of #8699 

---

#### What does this implement/fix? Explain your changes.

- Following #8842 implemented the torch-based MLP Classifier
- Added `MLPNetworkTorch`
- Created the requisite directory structure and renamed TensorFlow files

---

#### Does your contribution introduce a new dependency? If yes, which one?
No

---

#### What should a reviewer concentrate their feedback on?

- Use of the PyTorch library in the methods that build and train the model. 
- correct parameters handling for network and model construction.
- to check similar defaults used, as in TensorFlow counterpart.

---

#### Did you add any tests for the change?
No.

---

#### Any other comments?

No.

---

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://github.com/all-contributors/all-contributors/blob/master/docs/emoji-key.md)
- [x] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [x] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [x] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [x] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

File: sktime/networks/mlp/_mlp_tf.py [hunk 1]
```diff
@@ -65,6 +65,7 @@ def build_network(self, input_shape, **kwargs):
         input_layer = keras.layers.Input(input_shape)
```
Comment (RecreationalMath) [inline sktime/networks/mlp/_mlp_tf.py:87]: Lines 65 to 78 hardcode
- number of hidden layers in the network,
- number of neurons in each hidden layer,
- presence of dropout and dropout rate in hidden layers,
- presence of dropout and dropout rate in the output layer.

Usually, all of these are exposed to end users via parameters.
I haven't seen such enforcement in any other network so far. Is there a specific reason for this? @fkiraly

Comment (fkiraly) [reply sktime/networks/mlp/_mlp_tf.py:87]: I do not think there is any reason to not expose these to the user in some way. Probably just historically grown code.

Comment (RecreationalMath) [reply sktime/networks/mlp/_mlp_tf.py:87]: I also thought the same and wanted to confirm.

Comment (RecreationalMath) [reply sktime/networks/mlp/_mlp_tf.py:87]: > I do not think there is any reason to not expose these to the user in some way. Probably just historically grown code.

Upon further digging in, I found that this classifier is inspired by the model described in section `3.2.1 Multi Layer Perceptron` on page 14 in [Deep learning for time series classification: a review](https://arxiv.org/pdf/1809.04356) paper from 2019, and adapted from [this implementation](https://github.com/hfawaz/dl-4-tsc/blob/master/classifiers/mlp.py#L27C3-L44C71) of the same.

Do we want to stick with the same implementation as in the paper or expose the architecture-defining parameters to the end user and construct it accordingly? @fkiraly

Comment (RecreationalMath) [reply sktime/networks/mlp/_mlp_tf.py:87]: I've kept the settings from the paper as default values of the parameters, while exposing the parameters to the end user, providing flexibility to end user to modify the network setup as per their needs. 

The only parameter which would still differ is dropouts at each hidden layer, the original implementation has successively increasing dropouts at each layer, starting from `0.1` to `0.2` to `0.3`. Whereas we expose hidden dropouts as a single parameter, `dropout`, and use the same dropout value in each hidden layer.
 
Thoughts, @fkiraly ?

Comment (fkiraly) [reply sktime/networks/mlp/_mlp_tf.py:87]: Interesting - so, currently, we can only pass a single `dropout` parameter globally for other estimators where we can pass it, but MLP classifier has different dropouts by hidden layer?

The way I would solve it is:

* keep the option to allow a single float which means that dropout is set globally for all hidden layers
* add an option where dropout per layer can be passed, as a tuple of float (first is closest to input etc)

The default should be what the current default is, e.g., growing dropout rates by layer.

Comment (RecreationalMath) [reply sktime/networks/mlp/_mlp_tf.py:87]: > Interesting - so, currently, we can only pass a single `dropout` parameter globally for other estimators where we can pass it, but MLP classifier has different dropouts by hidden layer?

Yes.

> The way I would solve it is:
> 
> * keep the option to allow a single float which means that dropout is set globally for all hidden layers
> * add an option where dropout per layer can be passed, as a tuple of float (first is closest to input etc)
> 
> The default should be what the current default is, e.g., growing dropout rates by layer.

I second this.
Will push the change to implement this for `MLPNetworkTorch` in this PR and have opened #9103 for same to be implemented in all other deep learning models.

Comment (RecreationalMath) [reply sktime/networks/mlp/_mlp_tf.py:87]: > > The way I would solve it is:
> > 
> > * keep the option to allow a single float which means that dropout is set globally for all hidden layers
> > * add an option where dropout per layer can be passed, as a tuple of float (first is closest to input etc)
> > 
> > The default should be what the current default is, e.g., growing dropout rates by layer.
> 
> I second this. Will push the change to implement this for `MLPNetworkTorch` in this PR

Done.
